src/commit.c

```diff
@@ -279,15 +280,11 @@ int apk_solver_commit_changeset(struct apk_database *db,
 	foreach_array_item(change, changeset->changes) {
 		count_change(change, &prog.total);
 		if (change->new_pkg)
-			size_diff += change->new_pkg->installed_size / 1024;
+			size_diff += change->new_pkg->installed_size;
 		if (change->old_pkg)
-			size_diff -= change->old_pkg->installed_size / 1024;
-	}
-	size_unit = 'K';
-	if (labs(size_diff) > 10000) {
-		size_diff /= 1024;
-		size_unit = 'M';
+			size_diff -= change->old_pkg->installed_size;
 	}
+	size_unit = apk_get_human_unit(llabs(size_diff), &humanized);
```

Comment:
apk_get_human_unit() first argument is `size_t` so the `off_t` / `long long` gets silently casted down. The overflow happens now here.

Reply:
We could simply change the argument type of `apk_get_human_unit` to `off_t` or `long long` then. Regarding portability `double s = size` where `size` is an `off_t` could also overflow, couldn't it? So maybe it makes sense to also change the return value of `apk_get_human_unit` to `off_t` instead of using `double`?